controllers/progressiverollout_controller.go

```diff
@@ -34,15 +35,26 @@ type ProgressiveRolloutReconciler struct {
 	Scheme *runtime.Scheme
 }
 
-// +kubebuilder:rbac:groups=deployment.skyscanner.net.deployment.skyscanner.net,resources=progressiverollouts,verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups=deployment.skyscanner.net.deployment.skyscanner.net,resources=progressiverollouts/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups=deployment.skyscanner.net,resources=progressiverollouts,verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups=deployment.skyscanner.net,resources=progressiverollouts/status,verbs=get;update;patch
 
-func (r *ProgressiveRolloutReconciler) Reconcile(req ctrl.Request) (ctrl.Result, error) {
-	_ = context.Background()
-	_ = r.Log.WithValues("progressiverollout", req.NamespacedName)
+func (r *ProgressiveRolloutReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	log := r.Log.WithValues("progressiverollout", req.NamespacedName)
 
-	// your logic here
+	// Get the ProgressiveRollout object
+	pr := deploymentskyscannernetv1alpha1.ProgressiveRollout{}
+	if err := r.Get(ctx, req.NamespacedName, &pr); err != nil {
+		log.Error(err, "unable to fetch ProgressiveRollout", "object", pr.Name)
+		return ctrl.Result{}, client.IgnoreNotFound(err)
+	}
 
+	// Rollout completed
+	completed := pr.NewStatusCondition(deploymentskyscannernetv1alpha1.CompletedCondition, metav1.ConditionTrue, deploymentskyscannernetv1alpha1.StagesCompleteReason, "All stages completed")
```

Comment:
For now we simply set it to completed, is this right?

Reply:
Yeah, so at the moment we have just one type `Completed` that is `true` only when all the stages are completed.